Fix crash when beehive is broken by fake player

SquidDev opened the pull request:
When a beehive is broken, every nearby bee targets a random player. However, if there are no nearby players, the game crashes.

This should not occur under normal (vanilla) conditions. However, if a beehive is broken by a fake player (or potentially, a player with an extended reach attribute), there are no players in range and so we see a crash.

**Edit:** Apologies for the force-push spam. Used the wrong git config to commit.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/MinecraftForge/MinecraftForge?pullRequest=7566) <br/>All committers have signed the CLA.

SquidDev committed:
Fix crash when beehive is broken by fake player

When a beehive is broken, every nearby bee targets a random
player. However, if there are no nearby players, the game crashes.

This should not occur under normal (vanilla) conditions.
However, if a beehive is broken by a fake player (or potentially,
a player with an extended reach attribute), there are no players
in range and so we see a crash.

artheus reviewed (approved):
LGTM :+1:

LexManos committed:
Merge branch '1.16.x' into hotfix/beehive-no-player

LexManos committed:
Update BeehiveBlock.java.patch
